### ISSUE TITLE
Term needs more tests to document expected behavior, especially when value is a Resource

### ISSUE THREAD

**Issue by dchandekstark** [ActiveTriples/ActiveTriples#120]
Some methods of Term (at least in v0.2.3), such as #empty_property and #set do not appear to work properly when a Term value is a Resource.  For example, #empty_property only removes the statements having the term predicate rather than the values of the terms (the resources).  However, since neither the code not spec tests document the expected behavior I can't be sure this is bug.

**Comment by no-reply** [comment]
Can you post some examples of the behavior you aren't expecting? I'll try to take a look over the weekend.

**Comment by dchandekstark** [comment]
Here's a basic one, with an ActiveFedora 7 NtriplesRDFDatastream.  `<http://repository.lib.duke.edu/vocab/roles/hasRole>` is the predicate of the Term in this case, and you can see one resource (RDF type `<http://repository.lib.duke.edu/vocab/roles/Role>`).

```
_:g70221106533560 <http://repository.lib.duke.edu/vocab/roles/hasRole> _:g70221104847400 .
_:g70221104847400 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://repository.lib.duke.edu/vocab/roles/Role> .
_:g70221104847400 <http://repository.lib.duke.edu/vocab/roles/agent> "bob" .
_:g70221104847400 <http://repository.lib.duke.edu/vocab/roles/scope> "resource" .
_:g70221104847400 <http://repository.lib.duke.edu/vocab/roles/type> "Editor" .
```

After calling `empty_property` on the Term, only the statement with the Term predicate is removed, which is not of course what I want:

```
_:g70221104847400 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://repository.lib.duke.edu/vocab/roles/Role> .
_:g70221104847400 <http://repository.lib.duke.edu/vocab/roles/agent> "bob" .
_:g70221104847400 <http://repository.lib.duke.edu/vocab/roles/scope> "resource" .
_:g70221104847400 <http://repository.lib.duke.edu/vocab/roles/type> "Editor" .
```

**Comment by terrellt** [comment]
@dchandekstark @no-reply I'm interested in tightening up the piblic API for what is now Relation. 
Right now there's
```ruby
value_arguments=
clear
result
set
empty_property
build
first_or_create
delete
<<
[]=
property_config
type_property
reset! (which doesn't do anything...)
property
```

And then it delegates Array instance methods to #result (which is a fairly large API in and of itself.)

I see
```ruby
<<
[]=
build
delete (maybe? This is odd)
to_a (with array stuff delegated here maybe. Or it should be Enumerable itself, which uses this' #each method.)
```
as necessary

**Comment by no-reply** [comment]
I think I want to call this closed with #226. All of the public methods of `Relation` are now documented. Testing could still be improved, but I think we're pretty close to a coherent API.
